multicluster/controllers/multicluster/core/remote_cluster_manager.go

```diff
@@ -111,61 +109,55 @@ func (r *remoteClusterManager) Start() error {
 	klog.InfoS("Starting RemoteClusterManager", "ClusterSet", r.clusterSetID)
 	stopCtx, stopFunc := context.WithCancel(context.Background())
 	r.stopFunc = stopFunc
-	r.leaderElectionStopFunc, _ = r.StartLeaderElection()
-
-	go func() {
-		for {
-			select {
-			case <-stopCtx.Done():
-				klog.InfoS("Stopping RemoteClusterManager", "ClusterSet", r.clusterSetID)
-				r.mutex.Lock()
-				close(r.eventChan)
-				for _, rc := range r.remoteClusters {
-					rc.Stop()
+
+	// Start a Timer for every 5 seconds when leader election is
+	// performed, if necessary.
+	ticker := time.NewTicker(5 * time.Second)
+
+	for {
```

Comment:
Do you think better to keep the goroutine here, but remove the one to run Start() at the caller, so Start() and Stop() are consistent? No strong opinion. You should decide.

Reply:
Sure, I can do it along with the other changes in the followup PR

Reply:
Removed it in this PR.